### DISCUSSION TITLE
validation of subpresent

### DISCUSSION THREAD

**Discussion by NPhMKgbDNy1M** [renovatebot/renovate#15676] [Request Help] [answered]
How can I validate a subpresent?

renovate.json:

    {
     "$schema": "https://docs.renovatebot.com/renovate-schema.json",
     "extends": [
      "local>abc/def:default/subpresent-template"
    ]
    }

default.json:

    {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "subpresent-template": {
        "extends": [
            "config:base",
        ]
      },
      "subpresent-template2": {
        "extends": [
           "local>abc/def:default/subpresent-template",
           ":disableDependencyDashboard",
        ]
      },
    }

  Log:
  
    $ renovate-config-validator
    DEBUG: Using RE2 as regex engine
     INFO: Validating renovate.json
    DEBUG: Checking for config file in /builds/abc/def/default.json
     INFO: Validating default.json
    ERROR: default.json contains errors
           "errors": [
             {
               "topic": "Configuration Error",
               "message": "Invalid configuration option: subpresent-template"
             },
             {
               "topic": "Configuration Error",
               "message": "Invalid configuration option: subpresent-template2"
             }
           ]

**Comment by rarkins** [comment, marked as answer]
A subpreset is just a preset which another preset depends on, so it can be validated standalone like any regular preset.

I guess you have "sanitized" or simplified your example above but in doing so you've removed too much information for us to be able to know what's going on. For example template2 is missing altogether, and template has no content.

**Reply by NPhMKgbDNy1M** [reply]
Okay then I don't mean subpresent. If I understood correctly, in the template file (example: default.json) you can have multiple templates (example: subpresent-template, subpresent-template2). I have extended my example above.

Now I have the following problem. The custom json-top-level template names in default.json (example: subpresent-template) are not recognized by renovate-config-validator. The error message appears and the content of the individual template is not validated.

**Reply by NPhMKgbDNy1M** [reply]
The use-case of a validation like this is not that big for sure, but i would like to add it as a pipeline job. I was wondering if this is a bug?

**Reply by viceice** [reply]
we plan to deprecate sub-presets soon. so validator works as expected. 😉
